l10n_ar_perceptions_basic/models/perception.py

```diff
@@ -60,7 +61,47 @@ class PerceptionPerception(models.Model):
         ],
         default='nacional',
     )
-    account_analytic_id = fields.Many2one(
-        'account.analytic.account',
-        string='Analytic Account',
-    )
+    active = fields.Boolean('Active', default=True, store=True, readonly=False)
+
+    @api.multi
+    def unlink(self):
+        field_obj = self.env['ir.model.fields']
+        fields = field_obj.search([
+            ('relation', '=', 'perception.perception')
+        ])
+        for perception in self:
+            for field in fields:
+                model_obj = self.env[field.model]
+                if field.ttype == 'one2many':
+                    if not perception[field.relation_field]:
+                        continue
+                elif field.ttype == 'many2many':
+                    if field.relation_table:
+                        cr = self.env.cr
+                        query = """
+                            SELECT * FROM
+                        """
+                        query += field.relation_table
+                        query += """
+                            WHERE perception_id = %(perception)s
+                        """
+                        data = {
+                            'perception': perception.id
+                        }
+                        cr.execute(query, data)
+                        res = cr.fetchall()
+                        if not res:
+                            continue
+
+                    else:
+                        continue
+                else:
+                    res = model_obj.search([
+                        (field.name, '=', perception.id)
+                    ])
+                    if not res:
+                        continue
+                msg = 'Can not delete the record %s\
+                       because is used in the system.'
```

Comment:
```
msg = 'Can not delete the record %s\
                       because is used in the system.'
```
msg = ("Can not delete the record %s"
            "because is used in the system")

Is better because you avoid the extra spaces at the beginning of **because is used...**

Also a better error message would be appreciated, i.e. which relational field is causing the problem and which record is related, etc. _in the system_ is not very descriptive nor helpful, if nothing at all.